engine/apps/alerts/grafana_alerting_sync_manager/grafana_alerting_sync.py

```diff
@@ -102,50 +117,84 @@ def create_contact_points(self) -> None:
         # sync other datasource
         for datasource in datasources:
             if datasource["type"] == GrafanaAlertingSyncManager.ALERTING_DATASOURCE:
-                contact_point = self.create_contact_point(datasource)
+                contact_point, _ = self.create_contact_point(datasource)
                 if contact_point is None:
                     # Failed to create contact point duo to getting wrong alerting config. It is expected behaviour.
                     # Add datasource to list and retry to create contact point for it async
                     datasources_to_create.append(datasource)
 
         if datasources_to_create:
             logger.warning(
-                f"Some contact points were not created for integration {self.alert_receive_channel.pk}, "
-                f"trying to create async"
+                f"GrafanaAlertingSyncManager: Some contact points were not created for integration "
+                f"{self.alert_receive_channel.pk}, trying to create async"
             )
             # create other contact points async
             schedule_create_contact_points_for_datasource(self.alert_receive_channel.pk, datasources_to_create)
         else:
             self.alert_receive_channel.is_finished_alerting_setup = True
             self.alert_receive_channel.save(update_fields=["is_finished_alerting_setup"])
 
-    def create_contact_point(self, datasource=None) -> Optional["apps.alerts.models.GrafanaAlertingContactPoint"]:
+    def create_contact_point(
+        self, datasource=None
+    ) -> Tuple[Optional["apps.alerts.models.GrafanaAlertingContactPoint"], dict]:
         """
-        Try to create a contact point for datasource.
-        Return None if contact point was not created otherwise return contact point object
+        Update datasource config in Grafana Alerting and create OnCall contact point
         """
-        if datasource is None:
-            datasource = {}
+        if datasource is None:  # Means that we create contact point for default datasource
+            datasource = {
+                "type": GrafanaAlertingSyncManager.GRAFANA_CONTACT_POINT,
+            }
 
-        datasource_type = datasource.get("type") or GrafanaAlertingSyncManager.GRAFANA_CONTACT_POINT
         is_grafana_datasource = datasource.get("id") is None
+
+        # get alerting config
+        config, response_info, config_to_update = self.get_alerting_config_for_datasource(
+            datasource, is_grafana_datasource
+        )
+        if not config_to_update:  # failed to get alerting config for this datasource
+            return None, response_info
+
+        updated_config_from_alerting, response_info = self.add_contact_point_to_grafana_alerting(
+            datasource, is_grafana_datasource, config, config_to_update
+        )
+
+        if not updated_config_from_alerting:
+            return None, response_info
```

Comment:
Why we are returning response_info here? I didn't find any usage of second return value